pos_accounting/src/pool/storage/operator_impls.rs

```diff
@@ -301,7 +310,10 @@ impl<S: PoSAccountingStorageWrite<T>, T: StorageTag> PoSAccountingDB<S, T> {
 
         self.add_to_delegation_balance(undo_data.delegation_id, undo_data.amount)?;
 
-        self.add_balance_to_pool(pool_id, undo_data.amount)?;
+        // it's possible that the pool was decommissioned
```

Comment:
Decommissioning the pool should not prevent people from withdrawing their staked amounts.

Reply:
I'm trying to think if there's a corner case that isn't covered where this can be exploited.